x/vaultdecryption/vault_decrypter.go

```diff
@@ -0,0 +1,121 @@
+package vaultdecryption
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"github.com/cultureamp/ca-go/x/vaultdecryption/client"
+	"github.com/cultureamp/glamplify/log"
+	vaultapi "github.com/hashicorp/vault/api"
+)
+
+type Client interface {
+	RenewClient(ctx context.Context) error
+	GetSecret(batch []interface{}, keyReference string) (*vaultapi.Secret, error)
+}
+
+type VaultDecrypter interface {
+	Decrypt(keyReferences []string, encryptedData []string) ([]string, error)
+}
+
+type vaultDecrypter struct {
+	vaultClient Client
+	settings    *client.VaultSettings
+}
+
+func NewVaultDecrypter(vaultClient Client, settings *client.VaultSettings) *vaultDecrypter {
+	return &vaultDecrypter{vaultClient, settings}
+}
+
+func (v vaultDecrypter) Decrypt(keyReferences []string, encryptedData []string, ctx context.Context) ([]string, error) {
```

Comment:
`NewVaultDecrypter` returns a pointer of vaultDecrypter, so the methods on the struct should probably take in a pointer receiver.

```suggestion
func (v *vaultDecrypter) Decrypt(keyReferences []string, encryptedData []string, ctx context.Context) ([]string, error) {
```